### ISSUE TITLE
SSR not working with Gatsby

### ISSUE THREAD

**Issue by mcclaskiem** [mantinedev/mantine#1658]
### What package has an issue

@mantine/core

### Describe the bug

I have tried a bunch of different combinations and can't seem to get mantine to SSR correctly with Gatsby. The website is linked below:

https://splintersuite.com/

It looks like the overall layout of the website is getting SSR'd correctly but all the mantine components are flashing completely unstyled on page load. Below is my `gatsby-ssr.js`

```
import React from 'react';

import { renderToString } from 'react-dom/server';
import { createStylesServer, ServerStyles } from '@mantine/ssr';
import { renderStylesToString } from '@emotion/server';

import { ThemeProvider } from '@emotion/react';
import { MantineProvider } from '@mantine/core';

import theme from 'theme';

import Layout from 'src/layouts';

const stylesServer = createStylesServer();

export const wrapPageElement = ({ element }) => {
    return (
        <MantineProvider theme={theme}>
            <ThemeProvider theme={theme}>
                <Layout>{element}</Layout>
            </ThemeProvider>
        </MantineProvider>
    );
};
```
I am currently using the `gatsby-plugin-mantine` and `gatsby-plugin-emotion` however as you can see I have also tried manually setting up the `replaceRenderer` function. Is it possible that using emotion and mantine together would clash and break the SSR process somewhere along the way?

### In which browser did the problem occur

All

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Not sure

**Comment by rtivital** [comment]
> Is it possible that using emotion and mantine together would clash and break the SSR process somewhere along the way?

Most likely. Docs website is built with gatsby, you can always use it for reference – https://github.com/mantinedev/mantine/tree/master/docs

**Comment by mcclaskiem** [comment]
> > Is it possible that using emotion and mantine together would clash and break the SSR process somewhere along the way?
> 
> Most likely. Docs website is built with gatsby, you can always use it for reference – https://github.com/mantinedev/mantine/tree/master/docs

Do you know why Emotion would cause Mantine's SSR to break? Only Mantine is adding a `replaceRenderer` function, Emotion is doing some babel config in the `gatsby-node.js`

**Comment by rtivital** [comment]
Mantine uses custom emotion cache

**Comment by mcclaskiem** [comment]
Ok, as a test I just tried the following:

I removed emotion and changed over to `styled-components` and am still getting the same issue where any of the Mantine `<Text>` `<Title>` `<Button>` tags flash unstyled on page load. The overall page layout is there still on page load just any of the mantine components I have used flash unstyled.

Is there a different way we should be styling non-mantine JSX components? Ive built tons of Gatsby websites with styled components before and never had this issue that I am seeing with Mantine.
